numpy/random/mtrand/mtrand.pyx

```diff
@@ -1426,15 +1435,272 @@ cdef class RandomState:
         >>> plt.show()
 
         """
+        if sizeof(long) == sizeof(npy_int64):
+            return self.random_int64(low, high, size)
+        elif sizeof(long) == sizeof(npy_int32):
+            return self.random_int32(low, high, size)
+        else:
+            msg = 'Unimplemented for sizeof(long) == %d' % sizeof(long)
+            raise RuntimeError(msg)
+
+    def random_intp(self, low, high=None, size=None):
```

Comment:
You have a point, in that the way things are implemented the returned values would not change for the same ranges if only `np.intp` were used, but the array type would, doubling in memory footprint on windows, and the returned scalar values would be of `long` python type for python 2.7 on 64 bit Windows.